File: COM1786-CW1-Project-5/COM1786-CW1-Project-5/app/src/main/java/com/example/comp1786cw1project3/feature/homepage/HomeViewModel.java
package com.example.comp1786cw1project3.feature.homepage;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.comp1786cw1project3.feature.base.BaseViewModel;
import com.example.comp1786cw1project3.local.database.ExpenseDao;
import com.example.comp1786cw1project3.local.database.TripDao;
import com.example.comp1786cw1project3.model.Trip;

import java.util.List;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class HomeViewModel extends BaseViewModel {
    private MutableLiveData<List<Trip>> _trips = new MutableLiveData<>();
    LiveData<List<Trip>> trips = _trips;


    private TripDao tripDao;
    private ExpenseDao expenseDao;
    @Inject
    public HomeViewModel(TripDao tripDao, ExpenseDao expenseDao) {
        this.tripDao = tripDao;
        this.expenseDao = expenseDao;
        getTrips();
    }

    public void getTrips() {
        _trips.setValue(tripDao.getTrips());
    }

    public void resetDatabase() {
        tripDao.deleteAllTrip();
        expenseDao.deleteAllExpense();
        getTrips();
    }

    public void searchTrip(String key) {
        _trips.setValue(tripDao.searchTrip(key));
    }
}

File: COM1786-CW2-Project-5/COM1786-CW2-Project-5/app/src/main/java/com/example/comp1786cw2project2/feature/home/HomeViewModel.kt
package com.example.comp1786cw2project2.feature.home

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.comp1786cw2project2.feature.base.BaseViewModel
import com.example.comp1786cw2project2.local.database.UrlDao
import com.example.comp1786cw2project2.local.model.Url
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject


@HiltViewModel
class HomeViewModel @Inject constructor(
    private val urlDaoDatabase: UrlDao
) : BaseViewModel() {
    private var _currentUrl: MutableLiveData<Url> = MutableLiveData()
    var currentUrl : LiveData<Url> = _currentUrl
    private var currentIndex = 0

    fun storeImageToLocalDatabase(url: String) {
        urlDaoDatabase.addUrl(Url(url = url))
        val currentListUrl = urlDaoDatabase.getListUrl()
        currentIndex = currentListUrl.size - 1
        _currentUrl.value = Url(url = url)
    }

    fun getCurrentUrlImageFromDatabase(index: Int) {
        val listUrlImageFromDb = urlDaoDatabase.getListUrl()
        val range = listUrlImageFromDb.indices
        if (listUrlImageFromDb.isNotEmpty() && index in range) {
            _currentUrl.value = listUrlImageFromDb[index]
        }
    }

    fun onPreviousClicked() {
        val currentListUrl = urlDaoDatabase.getListUrl()
        val range = currentListUrl.indices
        currentIndex--
        if (currentListUrl.isNotEmpty() && currentIndex in range) {
            getCurrentUrlImageFromDatabase(currentIndex)
        } else {
            currentIndex++
        }
    }

    fun onNextClicked() {
        val currentListUrl = urlDaoDatabase.getListUrl()
        val range = currentListUrl.indices
        currentIndex++
        if (currentListUrl.isNotEmpty() && currentIndex in range) {
            getCurrentUrlImageFromDatabase(currentIndex)
        } else {
            currentIndex--
        }
    }
}